DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -1,369 +1,412 @@
+        protected List<InstallFile> PostUpgradeScripts
+        {
+            get
+            {
+                return this._postUpgradeScripts;
+            }
+        }
+
+        /// -----------------------------------------------------------------------------
+        /// <summary>
+        /// The Commit method finalises the Install and commits any pending changes.
+        /// </summary>
+        /// <remarks>In the case of Files this is not neccessary.</remarks>
+        /// -----------------------------------------------------------------------------
+        public override void Commit()
+        {
+            base.Commit();
+        }
+
+        /// -----------------------------------------------------------------------------
+        /// <summary>
+        /// The Install method installs the script component.
+        /// </summary>
+        /// -----------------------------------------------------------------------------
+        public override void Install()
+        {
+            this.Log.AddInfo(Util.SQL_Begin);
+            try
+            {
+                bool bSuccess = true;
+                Version installedVersion = this.Package.InstalledVersion;
+
+                // First process InstallScript
+                if (installedVersion == new Version(0, 0, 0))
+                {
+                    if (this.InstallScript != null)
+                    {
+                        bSuccess = this.InstallScriptFile(this.InstallScript);
+                        installedVersion = this.InstallScript.Version;
+                    }
+                }
+                else
+                {
+                    // Pre upgrade script
+                    foreach (InstallFile file in this.PreUpgradeScripts)
+                    {
+                        bSuccess = this.InstallScriptFile(file);
+                        if (!bSuccess)
+                        {
+                            break;
+                        }
+                    }
+                }
+
+                // Then process remain Install/Upgrade Scripts
+                if (bSuccess)
+                {
+                    foreach (InstallFile file in this.InstallScripts.Values)
+                    {
+                        if (file.Version > installedVersion)
+                        {
+                            bSuccess = this.InstallScriptFile(file);
+                            if (!bSuccess)
+                            {
+                                break;
+                            }
+                        }
+                    }
+                }
+
+                // Next process Post-UpgradeScripts - this script always runs if present
+                foreach (InstallFile file in this.PostUpgradeScripts)
+                {
+                    bSuccess = this.InstallScriptFile(file);
+                    if (file.Version != new Version(0, 0, 0))

```

Comment:
What's the purpose of requiring the file to have a version?  Are you wanting to check `installedVersion != new Version(0, 0, 0)`, i.e. it is an upgrade, not a new install?

Reply:
The old upgrade files could bump the version nr. I'm not sure if this is (a) something that would actually work as these run after the upgrade scripts and (2) something that would even be useful. In a next major upgrade we could remove this IMO.